lib/ansible/modules/remote_management/oneview/oneview_ethernet_network.py

```diff
@@ -0,0 +1,249 @@
+      name: 'Test Ethernet Network'
+      purpose: Management
+      bandwidth:
+          maximumBandwidth: 3000
+          typicalBandwidth: 2000
+  delegate_to: localhost
+
+- name: Ensure that the Ethernet Network is present with name 'Renamed Ethernet Network'
+  oneview_ethernet_network:
+    config: "{{ config_file_path }}"
+    state: present
+    data:
+      name: 'Test Ethernet Network'
+      newName: 'Renamed Ethernet Network'
+
+- name: Ensure that the Ethernet Network is absent
+  oneview_ethernet_network:
+    config: "{{ config_file_path }}"
+    state: absent
+    data:
+      name: 'New Ethernet Network'
+
+- name: Create Ethernet networks in bulk
+  oneview_ethernet_network:
+    config: "{{ config_file_path }}"
+    state: present
+    data:
+      vlanIdRange: '1-10,15,17'
+      purpose: General
+      namePrefix: TestNetwork
+      smartLink: false
+      privateNetwork: false
+      bandwidth:
+        maximumBandwidth: 10000
+        typicalBandwidth: 2000
+
+- name: Reset to the default network connection template
+  oneview_ethernet_network:
+    config: "{{ config_file_path }}"
+    state: default_bandwidth_reset
+    data:
+      name: 'Test Ethernet Network'
+  delegate_to: localhost
+'''
+
+RETURN = '''
+ethernet_network:
+    description: Has the facts about the Ethernet Networks.
+    returned: On state 'present'. Can be null.
+    type: dict
+
+ethernet_network_bulk:
+    description: Has the facts about the Ethernet Networks affected by the bulk insert.
+    returned: When 'vlanIdRange' attribute is in data argument. Can be null.
+    type: dict
+
+ethernet_network_connection_template:
+    description: Has the facts about the Ethernet Network Connection Template.
+    returned: On state 'default_bandwidth_reset'. Can be null.
+    type: dict
+'''
+
+from ansible.module_utils.oneview import OneViewModuleBase, OneViewModuleResourceNotFound
+
+
+class EthernetNetworkModule(OneViewModuleBase):
+    MSG_CREATED = 'Ethernet Network created successfully.'
+    MSG_UPDATED = 'Ethernet Network updated successfully.'
+    MSG_DELETED = 'Ethernet Network deleted successfully.'
+    MSG_ALREADY_PRESENT = 'Ethernet Network is already present.'
+    MSG_ALREADY_ABSENT = 'Ethernet Network is already absent.'
+
+    MSG_BULK_CREATED = 'Ethernet Networks created successfully.'
+    MSG_MISSING_BULK_CREATED = 'Some missing Ethernet Networks were created successfully.'
+    MSG_BULK_ALREADY_EXIST = 'The specified Ethernet Networks already exist.'
+    MSG_CONNECTION_TEMPLATE_RESET = 'Ethernet Network connection template was reset to the default.'
+    MSG_ETHERNET_NETWORK_NOT_FOUND = 'Ethernet Network was not found.'
+
+    RESOURCE_FACT_NAME = 'ethernet_network'
+
+    def __init__(self):
+
+        argument_spec = dict(
+            state=dict(
+                required=True,
+                choices=['present', 'absent', 'default_bandwidth_reset']
+            ),
+            data=dict(required=True, type='dict'),
```

Comment:
Make this:
```python
        argument_spec = dict(
            state=dict(type='str', required=True, choices=['absent', 'default_bandwidth_reset', 'present']),
            data=dict(type='dict', required=True),
        )
```
It's easier to read if there is one parameter per line, and we have 160 allowed columns on a line (and very wide screens these days).

I also sorted the options in the list (order is not important) and added the type as the first option (as this one should always be set anyway). So it looks nicer (to the eye of the beholder).